src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -692,6 +701,34 @@ def setUp(self):
                 manager_class_name, None, manager_subprocess, manager_mpi_type)
             self.set_job_manager_timeout()
 
+    def write_string_to_logfile(self, s):
+        """Write a string to the server log
+
+        Args:
+            s (string): string to write to log file.  Log message will be in
+                the following format:
+                  $date $hostname DAOS[$pid/0/6] rpc EMIT \
+                    src/cart/crt_rpc.c:53 crt_hdlr_ctl_log_add_msg() $s
+
+        """
+
+        # Fetch attachinfo data from server
+        attachinfo_file_path = self.agent_managers[0].get_attachinfo_file()
```

Comment:
Thinking about this a bit more, the way this is setup currently we will end up running `daos_agent dump_attachinfo` in parallel on each `DaosAgentManger.hosts` every time we want to log a message in the log file.  Does that make sense? Assuming the attach info is not going to change after the agent is started, would it make more sense to have the `DaosAgentManager` retain the attach info as a class property which is only obtained (running `daos_agent dump_attachinfo` in parallel on each host) if it is not already defined.  We could even define at the end of a successful `DaosAgentManager.start()` call - but keep in mind not all `DaosAgentManager` have their `start()` method called when we are continually running servers/agents across multiple test variants/methods, hence the property.

Reply:
> Thinking about this a bit more, the way this is setup currently we will end up running daos_agent dump_attachinfo in parallel on each DaosAgentManger.hosts every time we want to log a message in the log file. Does that make sense?

I don't think we'd thought of using this feature outside of `setUp()`.  We just want a test ID in the log file for mapping avocado variants to daos server logs.  When else are you thinking of writing messages to the daos server log?

Reply:
I would image it would it be useful to insert markers in the log when running tests like rebuild, osa, etc.